controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +205,128 @@ func (r *PolicyAutomationReconciler) Reconcile(
+			for clusterName, clusterEvent := range eventMap {
+				originalStartTime, err := time.Parse(time.RFC3339, clusterEvent.AutomationStartTime)
+				if err != nil {
+					log.Error(err, "Failed to retrieve AutomationStartTime in ClustersWithEvent")
+					delete(eventMap, clusterName)
+				}
+
+				preEventTime, err := time.Parse(time.RFC3339, clusterEvent.EventTime)
+				if err != nil {
+					log.Error(err, "Failed to retrieve EventTime in ClustersWithEvent")
+					delete(eventMap, clusterName)
+				}
+
+				// The time that delayAfterRunSeconds setting expires
+				delayUntil := originalStartTime.Add(time.Duration(delayAfterRunSeconds) * time.Second)
+
+				// The policy is non-compliant with the target cluster
+				if targetListMap[clusterName] {
+					// Policy status changed from non-compliant to compliant
+					// then back to non-compliant during the delay period
+					if delayAfterRunSeconds > 0 && preEventTime.After(originalStartTime) {
+						if now.After(delayUntil) {
+							// The delay period passed so remove the previous event
+							delete(eventMap, clusterName)
+							// Add the cluster name to create a new ansible job
+							trimmedTargetList = append(trimmedTargetList, clusterName)
+						} else {
+							requeueFlag = true
+							// Within the delay period and use the earliest requeueDuration to requeue
+							if (requeueDuration == 0) || (requeueDuration > int(delayUntil.Sub(now)+1)) {
+								requeueDuration = int(delayUntil.Sub(now) + 1)
+							}
+							// keep the event and update eventTime
+							clusterEvent.EventTime = nowStr
+							// new event from compliant to non-compliant
+							eventMap[clusterName] = clusterEvent
+						}
+					} // Otherwise, the policy keeps non-compliant since originalStartTime, do noting
+				} else { // The policy is compliant with the target cluster
+					if delayAfterRunSeconds > 0 && now.Before(delayUntil) {
+						// Within the delay period, keep the event and update eventTime
+						clusterEvent.EventTime = nowStr
+						// new event from non-compliant to compliant
+						eventMap[clusterName] = clusterEvent
+					} else { // No delay period or it is expired, remove the event
+						delete(eventMap, clusterName)
+					}
+				}
+			}
+
+			for _, clusterName := range targetList {
+				if _, ok := eventMap[clusterName]; !ok {
+					// Add the non-compliant clusters without previous automation event
+					trimmedTargetList = append(trimmedTargetList, clusterName)
+				}
+			}
+
+			if len(trimmedTargetList) > 0 {
+				log.Info("Creating An Ansible job", "trimmedTargetList", trimmedTargetList)
+				err = common.CreateAnsibleJob(
+					policyAutomation,
+					r.DynamicClient,
+					string(policyv1beta1.EveryEvent),
+					trimmedTargetList,
+				)
+
+				if err != nil {
+					log.Error(err, "Failed to create the Ansible job")
+
+					return reconcile.Result{}, err
+				}
+
+				automationStartTimeStr := time.Now().UTC().
+					Add(time.Second * time.Duration(1)).Format(time.RFC3339)
```

Comment:
Why are you adding 1 second here?

Reply:
The original `eventTime` is the same as `AutomationStartTime` on the second level, so during testing, I add 1 second to ensure `eventTime` is ealier, since we use `preEventTime.After(originalStartTime)`, I can remove `adding 1 second `